nvm.sh

```diff
@@ -1914,21 +1917,52 @@ nvm_install_binary() {
   if [ -f "${TARBALL}" ]; then
     TMPDIR="$(dirname "${TARBALL}")/files"
   fi
-  local tar
-  tar='tar'
-  if [ "${NVM_OS}" = 'aix' ]; then
-    tar='gtar'
+
+  local UNTAR_SUCCESS
+  UNTAR_SUCCESS=0
+
+  # For Windows system (GitBash with MSYS, Cygwin)
+  if [ "${NVM_OS}" = 'win' ]; then
+    if (
+      [ -n "${TMPDIR-}" ] && \
+      command mkdir -p "${TMPDIR}" && \
+      command unzip "${TARBALL}" -d "${TMPDIR}" && \
+      VERSION_PATH="$(nvm_version_path "${PREFIXED_VERSION}")/bin" && \
+      command mkdir -p "${VERSION_PATH}" && \
+      command mv "${TMPDIR}/"*/* "${VERSION_PATH}" && \
+      command rm -rf "${TMPDIR}"
+    ); then
+      UNTAR_SUCCESS=1
+    fi
+  # For non Windows system (including WSL running on Windows)
+  else
+    local tar_compression_flag
+    tar_compression_flag='z'
+    if nvm_supports_xz "${VERSION}"; then
+      tar_compression_flag='J'
+    fi
+
+    local tar
+    if [ "${NVM_OS}" = 'aix' ]; then
+      tar='gtar'
+    else
+      tar='tar'
+    fi
+
+    if (
+      [ -n "${TMPDIR-}" ] && \
+      command mkdir -p "${TMPDIR}" && \
+      command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" --strip-components 1 && \
```

Comment:
since this and the unzip command are the only difference between these two if blocks, can we restructure this so it's the same if, but with the extract command moved, perhaps, into a separate function that pivots based on the OS?

Reply:
I have tried to do it but it was less readable in the end, knowing that there 3 different things:

command unzip "${TARBALL}" -d "${TMPDIR}" && \
command "${tar}" -x${tar_compression_flag}f "${TARBALL}" -C "${TMPDIR}" --strip-components 1 && \

VERSION_PATH="$(nvm_version_path "${PREFIXED_VERSION}")/bin" && \
VERSION_PATH="$(nvm_version_path "${PREFIXED_VERSION}")" && \

command mv "${TMPDIR}/"*/* "${VERSION_PATH}" && \
command mv "${TMPDIR}/"* "${VERSION_PATH}" && \

The command, the VERSION_PATH with bin and the mv with extra /*

Reply:
I made nvm_install_binary_extract function to handle that